Tools/Scripts/webkitpy/port/darwin_testcase.py

```diff
@@ -40,6 +40,16 @@ def assert_skipped_file_search_paths(self, port_name, expected_paths, use_webkit
         port = self.make_port(port_name=port_name, options=MockOptions(webkit_test_runner=use_webkit2))
         self.assertEqual(port._skipped_file_search_paths(), expected_paths)
 
+    def test_sharding_groups(self):
+        port = self.make_port()
+        self.assertEqual(sorted(port.sharding_groups().keys()), ['media'])
+        self.assertEqual('media', port.group_for_shard('media/something'))
+        self.assertEqual('media', port.group_for_shard('webaudio/something'))
+        self.assertEqual('media', port.group_for_shard('fast/media/something'))
+        self.assertEqual('media', port.group_for_shard('media-session/something'))
+        self.assertEqual('media', port.group_for_shard('imported/mediacapture-fromelement/something'))
+        self.assertEqual(None, port.group_for_shard('fast/something'))
```

Comment:
Also `something/media` is probably worth testing here, so we know that behaviour there was deliberate.